### PR TITLE
Remove requirement to pass limit/offset when fetching logs

### PR DESCRIPTION
**Pull request by abnegate**
## What's Changed

Remove the requirement to pass a limit/offset for fetching logs. This makes it possible to request logs without passing a limit instead of using `PHP_INT_MAX` or using 0 to mean unlimited.

## Tests

- Existing tests

### CODE REVIEW

File: src/Audit/Audit.php [hunk 1]
```diff
@@ -189,21 +202,26 @@ public function log(string $userInternalId, string $userId, string $event, strin
...
-            $queries[] = Query::orderDesc('');
-            if ($orderAfter) {
+            if (! \is_null($limit)) {
+                $queries[] = Query::limit($limit);
+            }
+            if (! \is_null($offset)) {
+                $queries[] = Query::offset($offset);
+            }
+            if (! \is_null($orderAfter)) {
                 $queries[] = Query::cursorAfter($orderAfter);
```
Comment (fogelito) [inline src/Audit/Audit.php:225]: Is Cursor intentionally , not "Order"?

Comment (abnegate) [reply src/Audit/Audit.php:226]: Yep, `$orderAfter` is just the parameter name for the cursor document

File: src/Audit/Audit.php [hunk 2]
```diff
@@ -238,20 +260,27 @@ public function countLogsByUser(string $userId): mixed
...
+     * @param  int|null  $limit
+     * @param  int|null  $offset
      * @param  Document|null  $orderAfter
-     * @return mixed
+     * @return array<Document>
+     *
+     * @throws \Exception
      */
-    public function getLogsByResource(string $resource, int $limit = 25, int $offset = 0, Document $orderAfter = null): mixed
+    public function getLogsByResource(string $resource, ?int $limit = null, ?int $offset = null, ?Document $orderAfter = null): array
```
Comment (eldadfux) [inline src/Audit/Audit.php:270]: I don't think our problem is with the 25 default limit. It's actually good to have those defaults. We just don't need to throw errors if someone decides to use a much bigger number. We can still have support for `null` as unlimited. Let me know what you think

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: src/Audit/Audit.php (hunk 2 above)
Comment (abnegate) [reply src/Audit/Audit.php:270]: Ah yep makes sense! Will update all PR's

Comment (fogelito) [reply src/Audit/Audit.php:270]: If we have pagination does it make sense to have a max limit value?

**Review by eldadfux — APPROVED** [review]

File: src/Audit/Audit.php [hunk 3]
```diff
@@ -167,6 +168,11 @@ public function setup(): void
      * @param  string  $location
      * @param  array<string,mixed>  $data
      * @return bool
+     *
+     * @throws AuthorizationException
+     * @throws StructureException
+     * @throws \Exception
+     * @throws \Throwable
      */
     public function log(string $userInternalId, string $userId, string $event, string $resource, string $userAgent, string $ip, string $location, array $data = []): bool
```
Comment (eldadfux) [inline src/Audit/Audit.php:177]: Why do we have both userInternalId and userId here? this should be an Appwrite concept and utopia should be segmented from it. If we want the link to be by internalId, we should just pass that value to userId param, we don't need to maintain 2 variables.

**Review by eldadfux — CHANGES_REQUESTED** [review]

**Review by eldadfux — APPROVED** [review]

File: src/Audit/Audit.php (hunk 3 above)
Comment (abnegate) [reply src/Audit/Audit.php:177]: We will address it seperately 👌

**Pull request merged**